gallery/gallery-library/src/main/java/org/silverpeas/components/gallery/MediaUtil.java

```diff
@@ -212,445 +266,420 @@ public static void processPhoto(final FileHandler fileHandler, final Photo photo
-    final HandledFile toDir = getHandledFile(fileHandler, media).getParentHandledFile();
+    @Override
+    protected void generateFiles() {
 
-    // Copy and rename all media that exist into source folder
-    if (fromDir.exists()) {
+      final Photo photo = getMedia();
+      if (photo.isPreviewable()) {
 
-      // Copy all files which the name starts with the media identifier
-      Collection<HandledFile> srcFiles =
-          fromDir.listFiles(
-              or(
-                  prefixFileFilter(fromPK.getId()),
-                  asFileFilter((file) -> file.getName().matches(
-                      "^" + VIDEO_THUMBNAIL_FILE_PREFIX + "[0-9]+" +
-                          VIDEO_THUMBNAIL_FILE_EXTENSION + "$"))
-                  ),
-              falseFileFilter());
-      int substringIndex = fromPK.getId().length();
-      for (HandledFile srcFile : srcFiles) {
-        String srcFileName = srcFile.getFile().getName();
-        String dstFileName = (srcFileName.startsWith(fromPK.getId())) ?
-            (media.getId() + srcFileName.substring(substringIndex)) : srcFileName;
-        pasteFile(srcFile, toDir.getHandledFile(dstFileName), cut);
-      }
+        // Registering the size of the image
+        registerResolutionData();
 
-      // Copy original image
-      pasteFile(fromDir.getHandledFile(media.getFileName()), toDir.getHandledFile(media.
-          getFileName()), cut);
+        // Computing watermark data and retrieving the name of the author
+        final String nameForWatermark = computeWatermarkText();
 
-      // On cut operation, deleting the source repo
-      if (cut && !fromPK.getInstanceId().equals(media.getInstanceId())) {
+        // Creating preview and thumbnails
         try {
-          fromDir.delete();
-        } catch (Exception e) {
-          SilverLogger.getLogger(MediaUtil.class).error(
-              "Unable to delete source folder : folder path = " + fromDir.getFile().getPath(), e);
+          createThumbnails(nameForWatermark);
+        } catch (final Exception e) {
+          SilverLogger.getLogger(MediaUtil.class)
+              .error("image = " + photo.getTitle() + " (#" + photo.getId() + ")");
         }
       }
     }
-  }
 
-  private static void pasteFile(final HandledFile fromFile, final HandledFile toFile,
-      final boolean cut) {
-    if (fromFile.exists()) {
+    /**
+     * Registers the resolution of a photo.
+     */
+    private void registerResolutionData() {
+      if (getMedia().getDefinition().getWidth() != 0 &&
+          getMedia().getDefinition().getHeight() != 0) {
+        // definition already set.
+        return;
+      }
+      String[] widthAndHeight = null;
       try {
-        if (cut) {
-          fromFile.moveFile(toFile);
-        } else {
-          fromFile.copyFile(toFile);
+        widthAndHeight = getImageTool()
+            .getImageInfo(getHandledFile().getFile(), WIDTH_IN_PIXEL, HEIGHT_IN_PIXEL);
+      } catch (Exception ignore) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00108)